code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+        );
+
+        const quoteAssetFundsLpProvider1BeforeRemoval = await api.query.tokens.accounts(
+          walletLpProvider1.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+        const quoteAssetFundsLpProvider2BeforeRemoval = await api.query.tokens.accounts(
+          walletLpProvider2.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+        // 4. Remove initial liquidity
+        const lpTokenId = hardCodedPool1.lpTokenId;
+        const lpTokenAccountInfo1 = await api.query.tokens.accounts(walletLpProvider1.publicKey, lpTokenId);
+        const availableLpTokens1 = lpTokenAccountInfo1.free;
+
+        const removeAssetsMinAmounts = api.createType("BTreeMap<u128, u128>", {
+          "4": Pica(0),
+          "130": USDT(0)
+        });
+
+        const {
+          data: [resultWho, resultPoolId, resultReturnedAssetsAmount, resultMintedLp]
+        } = await sendAndWaitForSuccess(
+          api,
+          walletLpProvider1,
+          api.events.pablo.LiquidityRemoved.is,
+          // Parameters: poolId, assetsMap, minMintAmount, keepAlive
+          api.tx.pablo.removeLiquidity(hardCodedPool1.poolId, availableLpTokens1, removeAssetsMinAmounts) // ToDo: Update pool id w/ created pool!
+        );
+
+        // 5. Remove initial liquidity
+        const lpTokenAccountInfo2 = await api.query.tokens.accounts(walletLpProvider2.publicKey, lpTokenId);
+        const availableLpTokens2 = lpTokenAccountInfo2.free;
+
+        await sendAndWaitForSuccess(
+          api,
+          walletLpProvider2,
+          api.events.pablo.LiquidityRemoved.is,
+          // Parameters: poolId, assetsMap, minMintAmount, keepAlive
+          api.tx.pablo.removeLiquidity(hardCodedPool1.poolId, availableLpTokens2, removeAssetsMinAmounts) // ToDo: Update pool id w/ created pool!
+        );
+
+        const finalQuoteAssetFundsLpProvider1 = await api.query.tokens.accounts(
+          walletLpProvider1.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+        const finalQuoteAssetFundsLpProvider2 = await api.query.tokens.accounts(
+          walletLpProvider2.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+        expect(
+          finalQuoteAssetFundsLpProvider1.free
+            .sub(quoteAssetFundsLpProvider1BeforeRemoval.free)
+            .add(finalQuoteAssetFundsLpProvider2.free.sub(quoteAssetFundsLpProvider2BeforeRemoval.free))
+            .sub(new BN((100_000_000_000).toString()))
+        ).to.be.bignumber.closeTo(
+          new BN(amountEarnedFeesQuoteAssetWallet1.toString()).add(new BN(priceFirstTrade.toFixed(0))),
+          new BN(1)
         );
       }
     );
 
     it(
-      "Given that the users have sufficient balance," +
-        " users can't swap illegal token pairs(Non existing assetId in the pool)",
+      "#3.9 After adding funds, and after a trade, another wallet can add liquidity, withdraw it again without " +
```

Comment:
Perhaps a wording issue, but I think in this case second lp provider will receive a slightly less lp token based on the amounts in the pool, so the only difference would be that I guess.

Reply:
Good point, I'll update the test title be more precise.